client/core/bond.go

```diff
@@ -617,6 +628,216 @@ func (c *Core) repostPendingBonds(dc *dexConnection, cfg *dexBondCfg, state *dex
+		reusedBondedAmt += bond.Amount
+		if reusedBondedAmt >= mustPostAmt {
+			break
+		}
+	}
+
+	// If there are no bonds to update, then a new bond must be created.
+	if len(bondsToUpdate) == 0 {
+		return false, nil
+	}
+
+	var additionalAmtRequired uint64
+	if reusedBondedAmt < mustPostAmt {
+		additionalAmtRequired = mustPostAmt - reusedBondedAmt
+	}
+
+	amtToPost := additionalAmtRequired
+
+	// Make sure we're not posting more than MaxBondedAmt
+	if state.MaxBondedAmt > 0 {
+		expiredStrength := sumBondStrengths(state.ExpiredBonds, cfg.bondAssets)
+		currentlyBondedAmt := uint64(state.PendingStrength+state.LiveStrength+expiredStrength) * bondAsset.Amt
+		if currentlyBondedAmt >= state.MaxBondedAmt {
+			// If greater then something went wrong...
+			amtToPost = 0
+		} else if currentlyBondedAmt+amtToPost > state.MaxBondedAmt {
+			amtToPost = state.MaxBondedAmt - currentlyBondedAmt
+		}
+	}
+	amtToPost = amtToPost / bondAsset.Amt * bondAsset.Amt // round down to nearest bond size
+
+	// Make sure the new bond is not larger than we need
+	newBondSize := reusedBondedAmt + strongBondAmt + amtToPost
+	totalRequired := strongBondAmt + uint64(state.mustPost)*bondAsset.Amt
+	if totalRequired < newBondSize {
+		newBondSize = totalRequired
+	}
+
+	bondKeys := make([]*secp256k1.PrivateKey, 0, 2)
+	keyIndexes := make([]uint32, 0, 2)
+
+	// Will always create at least one new bond.
+	bondKey, keyIndex, err := c.nextBondKey(bondAsset.ID)
+	defer bondKey.Zero()
+	if err != nil {
+		return false, err
+	}
+	bondKeys = append(bondKeys, bondKey)
+	keyIndexes = append(keyIndexes, keyIndex)
+
+	// If we reusing more than the size of the new bond, we need
+	// a "change" bond.
+	if reusedBondedAmt+strongBondAmt > newBondSize {
+		bondKey, keyIndex, err := c.nextBondKey(bondAsset.ID)
+		defer bondKey.Zero()
+		if err != nil {
+			return false, err
+		}
+		bondKeys = append(bondKeys, bondKey)
+		keyIndexes = append(keyIndexes, keyIndex)
+	}
+
+	bondIDsToUpdate := make([]dex.Bytes, 0, len(bondsToUpdate))
+	for _, bond := range bondsToUpdate {
+		bondIDsToUpdate = append(bondIDsToUpdate, bond.CoinID)
+	}
+
+	newBonds, err := wallet.UpdateBondTx(bondAsset.Version, bondIDsToUpdate, bondKeys, newBondSize, lockTime)
+	if err != nil {
+		return false, codedError(bondPostErr, err)
+	}
+
+	nonChangeBond := newBonds[0]
+	if len(newBonds) == 2 {
+		nonChangeBond = newBonds[1]
+	}
+	err = c.preValidateBond(dc, nonChangeBond)
+	if err != nil {
+		return false, err
+	}
+
+	newDBBonds := make([]*db.Bond, 0, len(newBonds)+len(bondsToUpdate))
+	var replacementBondID dex.Bytes
+	for i, bond := range newBonds {
+		newDBBonds = append(newDBBonds, &db.Bond{
+			Version:    bond.Version,
+			AssetID:    bond.AssetID,
+			CoinID:     bond.CoinID,
+			UnsignedTx: bond.UnsignedTx,
+			SignedTx:   bond.SignedTx,
+			Data:       bond.Data,
+			Amount:     bond.Amount,
+			LockTime:   bond.LockTime,
+			KeyIndex:   keyIndexes[i],
+			RefundTx:   bond.RedeemTx,
```

Comment:
Yeah.. can't make one because we won't know what the nonce is when it's time to redeem. We could make a CLI tool to redeem your bonds in case something goes wrong.